src/as_list.cpp

```diff
@@ -0,0 +1,34 @@
+#include "r_trie.h"
+
+template <typename T>
+static inline Rcpp::List trie_to_list(SEXP radix){
+  r_trie <T>* rt_ptr = (r_trie <T> *) R_ExternalPtrAddr(radix);
+  ptr_check(rt_ptr);
+  
+  return Rcpp::List::create(
+    Rcpp::Named("keys") = rt_ptr->get_keys(),
```

Comment:
It's cool, I can't count the number of times I've overcomplicated something through language dorking (also, this is a good example of why I made sure they were class("trie") as well as "string_trie" etc. Foresight the current Oliver lacks)